### PR TITLE
Aqara Smart Plug device

### PR DESCRIPTION
**Pull request by sbkimaqara**

### CODE REVIEW

File: drivers/SmartThings/zigbee-switch/src/aqara/init.lua [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+local PREF_VALUE_MAX_POWER_DEFAULT = 23
+local PREF_VALUE_RESTORE_POWER_STATE_DEFAULT = false
+
+local LAST_REPORT_TIME = "LAST_REPORT_TIME"
+
+local FINGERPRINTS = {
+  { mfr = "LUMI", model = "lumi.plug.maeu01" }
+}
+
+local max_power_table = {
```
Comment (cjswedes) [inline drivers/SmartThings/zigbee-switch/src/aqara/init.lua:31]: I assume these values are Watts, but maybe there could be a comment or name change to confirm Also, could you comment on each line listing what that actual number is so a human can understand at first glance?

File: drivers/SmartThings/zigbee-switch/src/test/test_aqara_smart_plug.lua [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+      direction = "receive",
+      message = { mock_device.id,
+        SimpleMetering.attributes.CurrentSummationDelivered:build_test_attr_report(mock_device, 0x15a) }
+    },
+    {
+      channel = "capability",
+      direction = "send",
+      message = mock_device:generate_test_message("main",
+        capabilities.energyMeter.energy({ value = 0.346, unit = "kWh" }))
+    }
```
Comment (greens) [inline drivers/SmartThings/zigbee-switch/src/test/test_aqara_smart_plug.lua:208]: shouldn't there also be a PowerConsumptionReport event after this?

File: drivers/SmartThings/zigbee-switch/src/aqara/init.lua (hunk 1 above)
Comment (sbkimaqara) [reply drivers/SmartThings/zigbee-switch/src/aqara/init.lua]: Sure. I added a description.

File: drivers/SmartThings/zigbee-switch/src/test/test_aqara_smart_plug.lua (hunk 2 above)
Comment (sbkimaqara) [reply drivers/SmartThings/zigbee-switch/src/test/test_aqara_smart_plug.lua]: I added PowerConsumptionReport test code. :)

File: drivers/SmartThings/zigbee-switch/src/test/test_aqara_smart_plug.lua [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+    test.socket.zigbee:__expect_send({
+      mock_device.id,
+      zigbee_test_utils.build_bind_request(mock_device,
+        zigbee_test_utils.mock_hub_eui,
+        SimpleMetering.ID)
+    })
+    test.socket.zigbee:__expect_send({
+      mock_device.id,
+      SimpleMetering.attributes.InstantaneousDemand:configure_reporting(mock_device, 1, 3600, 1)
```
Comment (greens) [inline drivers/SmartThings/zigbee-switch/src/test/test_aqara_smart_plug.lua:120]: The defaults have been changed here in v45 which is why this test is failing.

Please test with the latest artifact and update this test accordingly: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/releases/tag/apiv2_45_1

Comment (Kwang-Hui) [reply drivers/SmartThings/zigbee-switch/src/test/test_aqara_smart_plug.lua:120]:
```suggestion
      SimpleMetering.attributes.InstantaneousDemand:configure_reporting(mock_device, 1, 3600, 5)
```

File: drivers/SmartThings/zigbee-switch/src/test/test_aqara_smart_plug.lua [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+    })
+    test.socket.zigbee:__expect_send({
+      mock_device.id,
+      zigbee_test_utils.build_bind_request(mock_device,
+        zigbee_test_utils.mock_hub_eui,
+        ElectricalMeasurement.ID)
+    })
+    test.socket.zigbee:__expect_send({
+      mock_device.id,
+      ElectricalMeasurement.attributes.ActivePower:configure_reporting(mock_device, 1, 3600, 1)
```
Comment (Kwang-Hui) [inline drivers/SmartThings/zigbee-switch/src/test/test_aqara_smart_plug.lua:134]:
```suggestion
      ElectricalMeasurement.attributes.ActivePower:configure_reporting(mock_device, 1, 3600, 5)
```

**Review by Kwang-Hui — COMMENTED** [review]
I run test on my side with change only this two values. it worked

File: drivers/SmartThings/zigbee-switch/src/test/test_aqara_smart_plug.lua (hunk 3 above)
Comment (sbkimaqara) [reply drivers/SmartThings/zigbee-switch/src/test/test_aqara_smart_plug.lua]: I updated. Thank you.

File: drivers/SmartThings/zigbee-switch/src/test/test_aqara_smart_plug.lua (hunk 4 above)
Comment (sbkimaqara) [reply drivers/SmartThings/zigbee-switch/src/test/test_aqara_smart_plug.lua]: I updated. Thank you.

File: drivers/SmartThings/zigbee-switch/src/aqara/init.lua [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+local write_pref_attribute = function(device, cluster_id, attribute_id, dt, value)
+  device:send(cluster_base.write_manufacturer_specific_attribute(device, cluster_id, attribute_id, MFG_CODE, dt, value))
+end
+
+local do_configure = function(self, device)
+  device:configure()
+
+  device:send(Groups.server.commands.RemoveAllGroups(device))
+end
```
[the remark this answers is not in the document]
Comment (greens) [reply drivers/SmartThings/zigbee-switch/src/aqara/init.lua:78]: that should let the defaults handle the attribute correctly

[the remark this answers is not in the document]
Comment (sbkimaqara) [reply drivers/SmartThings/zigbee-switch/src/aqara/init.lua:78]: Did I understand well?

```
local constants = require "st.zigbee.constants"

local do_configure = function(self, device)
  device:configure()
  device:set_field(constants.ELECTRICAL_MEASUREMENT_DIVISOR_KEY, 10)

  device:send(Groups.server.commands.RemoveAllGroups(device))
end
```

[the remark this answers is not in the document]
Comment (greens) [reply drivers/SmartThings/zigbee-switch/src/aqara/init.lua:78]: And delete the handler for `ActivePower`. Your tests for ActivePower should continue to be correct if it worked.

[the remark this answers is not in the document]
Comment (sbkimaqara) [reply drivers/SmartThings/zigbee-switch/src/aqara/init.lua:78]: 😀 Is this OK? 

```
local constants = require "st.zigbee.constants"

local do_configure = function(self, device)
  device:configure()
  device:set_field(constants.ELECTRICAL_MEASUREMENT_DIVISOR_KEY, 10)

  device:send(Groups.server.commands.RemoveAllGroups(device))
end

local aqara_smart_plug_handler = {
  NAME = "Aqara Smart Plug Handler",
  lifecycle_handlers = {
    added = device_added,
    doConfigure = do_configure,
    infoChanged = device_info_changed
  },
  zigbee_handlers = {
    attr = {
      [SimpleMetering.ID] = {
        [SimpleMetering.attributes.InstantaneousDemand.ID] = power_meter_handler,
        [SimpleMetering.attributes.CurrentSummationDelivered.ID] = energy_meter_handler
      }
    }
  },
  can_handle = is_aqara_products,
}
```

[the remark this answers is not in the document]
Comment (sbkimaqara) [reply drivers/SmartThings/zigbee-switch/src/aqara/init.lua:78]: Thank you for your help 😀

**Review by greens — APPROVED** [review]

**Pull request merged**